spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/ReorderMemory.scala

```diff
@@ -84,6 +85,11 @@ class ReorderMemory(inputFormat: MemoryData, outputFormat: MemoryData,
 
     // recover to original data
     output = initTensor(realOutput(0))
+
+    if (outputLayout == Memory.Format.nhwc && realOutput(0).isInstanceOf[HeapData]) {
+      output.toTensor[Float].resize(_outputFormats(0).shape)
```

Comment:
if not heapdata, output may not have real size, is it right?

Reply:
The tensor size will always be right.